src/pages/pretest-checks/pretest-checks.html

```diff
@@ -3,112 +3,140 @@
-          Instructor reg.
-        </ion-col>
-        <ion-col col-8>
-          <input type="text" class="mes-text-input">
-        </ion-col>
-      </ion-row>
-      <div class="mes-divider"></div>
-      <ion-row>
-        <ion-col col-4 class="mes-header-md">
-          Instructor cert.
-        </ion-col>
-        <ion-col col-8>
-          <input type="text" class="mes-text-input">
-        </ion-col>
-      </ion-row>
-    </ion-card-content>
-  </ion-card>
-  <div class="action-buttons-wrapper">
-    <button class="mes-button-danger end-test-button" [navPush]="endTestReasonPage">End Test</button>
-    <button class="mes-button-primary confirm-button" (click)="gotoDL25()">Confirm</button>
-    <div class="clearfix"></div>
-  </div>
+    <ion-card class="mes-card card-top-separator">
+      <ion-card-content>
+        <ion-row>
+          <ion-col col-4 class="mes-header-md">
+            Car registration*
+          </ion-col>
+          <ion-col col-8>
+            <input required type="text" class="mes-text-input capital-characters" #carRegistration="ngModel" name="carRegistration" [(ngModel)]="preCheck.carRegistration"
+              autocapitalize="characters">
+          </ion-col>
+        </ion-row>
+        <p [hidden]="carRegistration.valid || (carRegistration.pristine && !form.submitted)" class="warning-text">
+          Car registration number is required.
+        </p>
+        <div class="mes-divider"></div>
+        <ion-row class="height-auto">
+          <ion-col col-4 class="mes-header-md">
+            Automatic*
+            <input type="checkbox" hidden required #isAutomatic="ngModel" name="isAutomatic" [(ngModel)]="preCheck.isAutomatic">
+          </ion-col>
+          <ion-col col-4 class="mes-header-md">
+            Yes
+            <input #yesInput type="checkbox" class="mes-checkbox" (change)="automaticInputChanged($event, noInput);">
+          </ion-col>
+          <ion-col col-4 class="mes-header-md">
+            No
+            <input #noInput type="checkbox" class="mes-checkbox" (change)="automaticInputChanged($event, yesInput);">
+          </ion-col>
+        </ion-row>
+        <p [hidden]="isAutomatic.valid || (isAutomatic.pristine && !form.submitted)" class="warning-text">
+          One option must be selected.
+        </p>
+        <div class="mes-divider"></div>
+        <ion-row class="center-text">
+          <ion-col col-6 class="mes-header-md">
+            School car
+            <input type="checkbox" class="mes-checkbox">
+          </ion-col>
+          <ion-col col-6 class="mes-header-md">
+            D/C
+            <input type="checkbox" class="mes-checkbox">
+          </ion-col>
+        </ion-row>
+        <div class="mes-divider"></div>
+        <ion-row>
+          <ion-col col-4 class="mes-header-md">
+            Instructor reg.
+          </ion-col>
+          <ion-col col-8>
+            <input type="number" pattern="[0-9]*" class="mes-text-input">
```

Comment:
It might be worth noting that number fields still allow the user to enter floating point numbers (https://stackoverflow.com/a/31706796), including the use of the letter e. I don't think it matters for Alpha but it might be something we should be aware of for Beta.

Reply:
yep, I mentioned this issue in US. There are 2 solutions: 1. Add validation (some regex) 2. Build our own keyboard...